### PR TITLE
Added Yahoo search terms to browser_search, #1743

### PR DESCRIPTION
**Pull request by kiddinn**
## One line description of pull request

Adding Yahoo search terms to the browser_search analysis plugin.

## Description:

**Related issue (if applicable):** fixes #<plaso issue number here>

## Notes:
All contributions to Plaso undergo [code
review](https://github.com/log2timeline/l2tdocs/blob/master/process/Code%20review%20process.asciidoc). This makes sure
that the code has appropriate test coverage and conforms to the Plaso [style
guide](https://github.com/log2timeline/plaso/wiki/Style-guide).

One of the maintainers will examine your code, and may request changes. Check off the items below in
order, and then a maintainer will review your code.

## Checklist:
  - [ ] Automated checks (Travis, Codecov, Codefactor )pass
  - [ ] No new [new dependencies](https://github.com/log2timeline/plaso/wiki/Adding-a-new-dependency) are required or l2tdevtools has been updated
  - [ ] Reviewer assigned

### CODE REVIEW

**Comment by joachimmetz** [comment]
MacOS test is failing with:
```
======================================================================
ERROR: testExtractEventsFromSourcePartitionedImage (cli.psteal_tool.PstealToolTest)
Tests the ExtractEventsFromSources function on a partitioned image.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/travis/build/log2timeline/plaso/tests/cli/psteal_tool.py", line 307, in testExtractEventsFromSourcePartitionedImage
    test_tool.ExtractEventsFromSources()
  File "./plaso/cli/psteal_tool.py", line 345, in ExtractEventsFromSources
    status_update_callback=status_update_callback)
  File "./plaso/multi_processing/task_engine.py", line 827, in ProcessSources
    filter_find_specs=filter_find_specs)
  File "./plaso/multi_processing/task_engine.py", line 346, in _ProcessSources
    self._ScheduleTasks(storage_writer)
  File "./plaso/multi_processing/task_engine.py", line 452, in _ScheduleTasks
    self._MergeTaskStorage(storage_writer)
  File "./plaso/multi_processing/task_engine.py", line 261, in _MergeTaskStorage
    maximum_number_of_containers=self._MAXIMUM_NUMBER_OF_CONTAINERS)
  File "./plaso/storage/sqlite/merge_reader.py", line 240, in MergeAttributeContainers
    self._PrepareForNextContainerType()
  File "./plaso/storage/sqlite/merge_reader.py", line 210, in _PrepareForNextContainerType
    self._cursor.execute(query)
OperationalError: disk I/O error
```

Running it again

File: plaso/analysis/browser_search.py [hunk 1]
```diff
@@ -57,6 +57,8 @@ class BrowserSearchPlugin(interface.AnalysisPlugin):
        '_ExtractGoogleSearchQuery'),
       ('Google Sites', re.compile(r'sites.google.com/site'),
```
Comment (joachimmetz) [inline plaso/analysis/browser_search.py]: I know this is not related to your changes, but could you update `sites.google.com` as well to `sites\.google\.com` and check if there are more URL without proper escaping. Thx in advance.

File: plaso/analysis/browser_search.py [hunk 2]
```diff
@@ -175,6 +177,28 @@ def _ExtractGoogleSearchQuery(self, url):
 
     return line.replace('+', ' ')
 
+  def _ExtractYahooSearchQuery(self, url):
+    """Extracts a search query from a Yahoo search URL.
+
+    Yahoo: https://search.yahoo.com/search?p=query
```
Comment (joachimmetz) [inline plaso/analysis/browser_search.py]: Please remove/reword 'Yahoo: ' I have no idea what you're telling me with this.

Maybe:
```
Examples:
  https://search.yahoo.com/search?p=query
  ...
```

File: plaso/analysis/browser_search.py [hunk 3]
```diff
@@ -175,6 +177,28 @@ def _ExtractGoogleSearchQuery(self, url):
 
     return line.replace('+', ' ')
 
+  def _ExtractYahooSearchQuery(self, url):
+    """Extracts a search query from a Yahoo search URL.
```
Comment (joachimmetz) [inline plaso/analysis/browser_search.py]: Technically this function extract a search query from any URL that contains `q=`. Is this the intended functionality? If so maybe reflect this in the docstring, and maybe this method can be merged with other methods that do the same?

Comment (joachimmetz) [reply plaso/analysis/browser_search.py]: I'm OK with pushing now, if this is urgently needed and create and issue and follow up CL to generalize this.

Comment (joachimmetz) [reply plaso/analysis/browser_search.py]: I've created: https://github.com/log2timeline/plaso/issues/2216

**Comment by joachimmetz** [comment]
Need to fix tests before merging

**Comment by kiddinn** [comment]
it still seems like all plaso tests are succeeding ,it's just dftimewolf and l2tbinaries that are failing

**Comment by joachimmetz** [comment]
> it still seems like all plaso tests are succeeding ,it's just dftimewolf and l2tbinaries that are failing

https://github.com/log2timeline/plaso/commit/c7c0411d127503e404c3327e1a994c775dfb6727 fixed the tests and I re-started them yesterday before going offline. All green now

**Pull request merged**